pydatastructs/graphs/algorithms.py

```diff
@@ -877,3 +879,73 @@ def _job(graph: Graph, u: str):
     if len(L) != num_vertices:
         raise ValueError("Graph is not acyclic.")
     return L
+    
+def minDistance(graph: Graph,dist,visited,source):
+    min = sys.maxsize
+    min_index=source
+    for v in graph.vertices:
+        if dist[v] < min and visited[v] is False:
+            min = dist[v] 
+            min_index = v 
+    return min_index 
+        
+def dijkstra_algorithm(graph: Graph,start: str):
+    """
+    Finds shortest path distance in the given graph from a given source to all vertex.
+
+    Parameters
+    ==========
+
+    graph: Graph
+        The graph under consideration.
+    start: str
+        The name of the source the node.
+    
+    Returns
+    =======
+
+    Displays Vertex and Distance from Source as a Key value pair
+    
+    Examples
+    ========
+
+    >>> from pydatastructs import Graph, AdjacencyListGraphNode
+    >>> from pydatastructs import shortest_paths
+    >>> V1 = AdjacencyListGraphNode("V1")
+    >>> V2 = AdjacencyListGraphNode("V2")
+    >>> V3 = AdjacencyListGraphNode("V3")
+    >>> G = Graph(V1, V2, V3)
+    >>> G.add_edge('V2', 'V3', 14)
+    >>> G.add_edge('V1', 'V2', 12)
+    >>> dijkstra_algorithm(G,'V1')
+    {'V1': 0, 'V2': 12, 'V3': 26}
+    
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Dijkstra%27s_algorithm#:~:text=Dijkstra's
+    """
+    V=len(graph.vertices)
+    visited={}
+    dist={}
+    for v in graph.vertices:
+        visited[v]=False
+        if v!=start:
+            dist[v] =sys.maxsize
+    dist[start] = 0
+        
+    for cout in range(V):
+        u = minDistance(graph,dist,visited,start) 
+        visited[u] = True
+        for v in graph.vertices: 
+            try:
+                if graph.edge_weights[u+"_"+v].value> 0 and visited[v] is False and dist[v] > dist[u] + graph.edge_weights[u+"_"+v].value:
+                    dist[v] = dist[u] + graph.edge_weights[u+"_"+v].value
+            except:
+                pass
```

Comment:
```suggestion
    V = len(graph.vertices)
    visited = {}
    dist = {}
    for v in graph.vertices:
        visited[v] = False
        if v != start:
            dist[v] = float('inf')
    dist[start] = 0
        
    for cout in range(V):
        u = min_distance(graph, dist, visited, start) 
        visited[u] = True
        for v in graph.vertices: 
            try:
                if graph.edge_weights[u+"_"+v].value > 0 and visited[v] is False and dist[v] > dist[u] + graph.edge_weights[u+"_"+v].value:
                    dist[v] = dist[u] + graph.edge_weights[u+"_"+v].value
            except:
                pass
```
Why we need `try-except` block here?

Reply:
this what meant to do?

Reply:
> this what meant to do?

This probably does what you wanted to do but without `try-except`. `try-except` shouldn't be used usually.